dom-event-delegation/main.ts

```diff
@@ -0,0 +1,24 @@
+document.addEventListener('DOMContentLoaded', () => {
+  const taskList = document.querySelector('.task-list');
+
+  if (!taskList) {
+    throw new Error('Task list element not found');
+  }
+
+  taskList.addEventListener('click', (event: Event) => {
+    const eventTarget = event.target as HTMLElement;
+    console.log('Event Target:', eventTarget);
+    console.log('Tag Name:', eventTarget.tagName);
+    if (
+      eventTarget.tagName === 'BUTTON' &&
+      eventTarget.closest('.task-list-item')
+    ) {
+      const taskListItem = eventTarget.closest(
+        '.task-list-item'
+      ) as HTMLElement;
+      console.log(taskListItem);
+      taskListItem.remove();
+      console.log(eventTarget.tagName);
+    }
+  });
```

Comment:
I would recommend using "guard clauses." It essentially checks if _not_ the value we want and allows us to early exit out of the function. It makes the function a tiny bit more efficient but it makes it much more readable and easy to understand.
```
if ($eventTarget.tagName !== 'BUTTON') {
    return;
  }

  const $taskListItem = $eventTarget.closest('.task-list-item');

  console.log('closest .task-list-item', $taskListItem);
  $taskListItem?.remove();
```